pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -76,55 +126,44 @@ type manager struct {
 	certAccessLock           sync.RWMutex
 	cert                     *tls.Certificate
 	shouldRotatePercent      uint
+	forceRotation            bool
 }
 
 // NewManager returns a new certificate manager. A certificate manager is
 // responsible for being the authoritative source of certificates in the
 // Kubelet and handling updates due to rotation.
-func NewManager(
-	certSigningRequestClient certificatesclient.CertificateSigningRequestInterface,
-	template *x509.CertificateRequest,
-	usages []certificates.KeyUsage,
-	certificateStore Store,
-	certRotationPercent uint) (Manager, error) {
-
-	cert, err := certificateStore.Current()
+func NewManager(config *Config) (Manager, error) {
+	cert, forceRotation, err := getCurrentCertificateOrBootstrap(
```

Comment:
Because there may not be a valid reference to a client yet. There is one particular use case that makes this whole initialization awkward. When the REST client that connects to the Certificate Signing API starts, it needs the certificate information from the certificate manager (which has to find the most recent cert/key pair, or disambiguate different boot strap situations). When the certificate manager rotation starts, it needs the REST client connection to the API server.